src/devices/Bno055/Bno055Sensor.cs

```diff
@@ -123,24 +123,37 @@ public Bno055Sensor(I2cDevice i2cDevice, OperationMode operationMode = Operation
             Info.FirmwareVersion = new Version(ReadByte(Registers.SW_REV_ID_MSB), ReadByte(Registers.SW_REV_ID_LSB));
             Info.BootloaderVersion = new Version(ReadByte(Registers.BL_REV_ID), 0);
 
-            // Check if a reset is needed
-            if (TemperatureSource != TemperatureSource.Gyroscope)
-            {
-                WriteReg(Registers.SYS_TRIGGER, 0x20);
-                // Need 650 milliseconds after reset
-                Thread.Sleep(650);
-                PowerMode = PowerMode.Normal;
-                WriteReg(Registers.SYS_TRIGGER, 0x00);
-                TemperatureSource = TemperatureSource.Gyroscope;
+            _operationMode = operationMode;
+            initializeRegisters();
+        }
+
+        private static readonly byte[][] registerDefaults =
+        {
+            new byte[]{ (byte)Registers.UNIT_SEL, (byte)(Units.AccelerationMeterPerSecond | Units.AngularRateDegreePerSecond | Units.DataOutputFormatWindows | Units.EulerAnglesDegrees | Units.TemperatureCelsius) },
```

Comment:
nit: space after `[]`

also please try to split long lines into multiple

Reply:
I wish, Roslyn would auto-format those...